dask_xgboost/core.py

```diff
@@ -187,6 +195,12 @@ def train(client, params, data, labels, dmatrix_kwargs={}, **kwargs):
     --------
     predict
     """
+    if (client is None or
+            not dask.is_dask_collection(data) or
+            not dask.is_dask_collection(labels)):
+        dtrain = xgb.DMatrix(data, labels, **dmatrix_kwargs)
+        return xgb.train(params, dtrain, **kwargs)
```

Comment:
We could also raise here and point the user to `xgb.train` with an informative message.

Reply:
I'm still debugging this, but it seems like in a `Randomized/GridSearchCV` context we can end up with a concrete ndarray here.

Reply:
I think https://github.com/dask/dask-xgboost/pull/28/files#r214392082 left out the important bit of information that GridSearchCV can pass a concrete ndarary here, *even if you start with a dask array*.